src/api/routes/bns/pricing.ts

```diff
@@ -0,0 +1,138 @@
+      res.status(400).json({ error: 'Invalid namespace' });
+      return;
+    }
+    const randomPrivKey = makeRandomPrivKey();
+    const address = getAddressFromPrivateKey(
+      randomPrivKey.data,
+      chainId === ChainID.Mainnet ? TransactionVersion.Mainnet : TransactionVersion.Testnet
+    );
+    const bnsContractIdentifier = getBnsContractID(chainId);
+    if (!bnsContractIdentifier || !isValidPrincipal(bnsContractIdentifier)) {
+      logger.error('BNS contract ID not properly configured');
+      return res.status(500).json({ error: 'BNS contract ID not properly configured' });
+    }
+
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: address,
+      contractAddress: bnsContractIdentifier,
+      contractName: 'bns',
+      functionName: 'compute-namespace-price?',
+      functionArgs: [bufferCVFromString(namespace)],
+      network: stacksNetwork,
+    };
+    try {
+      const contractCallTx = await callReadOnlyFunction(txOptions);
+      if (
+        contractCallTx.type == ClarityType.ResponseOk &&
+        contractCallTx.value.type == ClarityType.UInt
+      ) {
+        const response: BnsGetNamespacePriceResponse = {
+          units: 'STX',
+          amount: contractCallTx.value.value.toString(10),
+        };
+        res.json(response);
+      } else {
+        res.status(400).json({ error: 'Invalid namespace' });
+      }
+    } catch (error) {
+      res.status(400).json({ error: 'Error calling readOnlyFunction' });
+    }
+  });
+
+  router.getAsync('/names/:name', async (req, res) => {
+    const input = req.params.name;
+    if (!input.includes('.')) {
+      res.status(400).json({ error: 'Invalid name' });
+      return;
+    }
+    const split = input.split('.');
+    if (split.length != 2) {
+      res.status(400).json({ error: 'Invalid name' });
+      return;
+    }
+    const name = split[0];
+    const namespace = split[1];
+    const namespaceQuery = await db.getNamespace({ namespace });
+    if (!namespaceQuery.found) {
+      res.status(400).json({ error: 'Namespace not exits' });
+      return;
+    }
+    const dbNamespace: DbBnsNamespace = namespaceQuery.result;
+    const randomPrivKey = makeRandomPrivKey();
+    const address = getAddressFromPrivateKey(
+      randomPrivKey.data,
+      chainId === ChainID.Mainnet ? TransactionVersion.Mainnet : TransactionVersion.Testnet
+    );
+    const buckets = dbNamespace.buckets.split(';').map(x => uintCV(x));
+
+    const bnsContractIdentifier = getBnsContractID(chainId);
+    if (!bnsContractIdentifier || !isValidPrincipal(bnsContractIdentifier)) {
+      logger.error('BNS contract ID not properly configured');
+      return res.status(500).json({ error: 'BNS contract ID not properly configured' });
+    }
+
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: address,
+      contractAddress: bnsContractIdentifier,
+      contractName: 'bns',
+      functionName: 'compute-name-price',
```

Comment:
This function no longer exists, we should be using `get-name-price`

Reply:
Fixed in 5a84407 and 291b74c